### PR TITLE
Added every type of salt function validation

### PR DESCRIPTION
**Pull request by spacanowski**
Not not only ```cmd.script``` functions will have their outcome validated/checked.

### CODE REVIEW

**Comment by mchugh19** [comment]
Thanks for working this up. I was away last week, but have returned so I'll get to looking it over.

File: src/main/java/com/waytta/SaltAPIBuilder.java [hunk 1]
```diff
@@ -47,515 +44,561 @@
...
+            // Add any args to json message
+            saltFunc.element("arg", saltArguments);
+        }
+    }
 
     /*
      * We'll use this from the <tt>config.jelly</tt>.
      */
     public String getServername() {
-	return servername;
```
Comment (mchugh19) [inline src/main/java/com/waytta/SaltAPIBuilder.java]: Let's leave the get methods up above the constructor. There's a reformatting I'd like to see happen that removes the arguments from the constructor in favor of set calls. If/when that goes through, these should all be near the @DataBoundConstructor

File: src/main/java/com/waytta/SaltAPIBuilder.java [hunk 2]
```diff
@@ -47,515 +44,561 @@
...
+            if (returnArray.get(0).toString().contains("TypeError")) {
+                listener.getLogger().println("Salt reported an error for " + myfunction + " "
+                        + myarguments + " for " + mytarget + ":\n" + returnArray.toString(2));
+                return false;
+            }
+
+            // If running script module check for non-zero exit codes in
+            // retcode.
+            returnArray = httpResponse.getJSONArray("return");
+            boolean validFunctionExecution = Utils.validateFunctionCall(returnArray);
```
Comment (mchugh19) [inline src/main/java/com/waytta/SaltAPIBuilder.java]: I'm not in front of a salt environment at the moment. Has this bit been tested, that all functions set a proper return code? 
test.ping
test.arg
cmd.run
cmd.script
state.sls
Plus the salt runner for state.orchestrate

File: src/main/java/com/waytta/SaltAPIBuilder.java (hunk 1 above)
Comment (spacanowski) [inline src/main/java/com/waytta/SaltAPIBuilder.java]: Done

File: src/main/java/com/waytta/SaltAPIBuilder.java (hunk 2 above)
Comment (spacanowski) [inline src/main/java/com/waytta/SaltAPIBuilder.java]: This was modified so in my work we get full CI with saltstack execution checking. We use ```state.sls``` and when it fails it returns ```retcode```. I did it so it's checked if there is ```retcode``` in json tree. If not it's ignored (proper execution), if it's present we check this code. Added some tests to show usage for different json structures and occurrence and lack of occurrence of ```retcode```.

**Pull request merged**